src/main/java/com/indeed/jiraactions/Paginator.java

```diff
@@ -67,16 +69,21 @@ public void process() throws InterruptedException {
                 final Stopwatch stopwatch = Stopwatch.createStarted();
                 final List<Issue> issues = Lists.newArrayList(pageProvider.getPage());
                 log.debug(String.join(", ", issues.stream().map(x -> x.key).collect(Collectors.toList())));
-                for(final Issue issue : issues) {
+                for (final Issue issue : issues) {
                     try {
                         final List<Action> preFilteredActions = pageProvider.getActions(issue);
                         final List<Action> actions = getActionsFilterByLastSeen(seenIssues, issue, preFilteredActions);
                         final List<Action> filteredActions = actions.stream().filter(a -> a.isInRange(startDate, endDate)).collect(Collectors.toList());
+
+                        if (buildJiraIssues && !filteredActions.isEmpty()) {
```

Comment:
Could have sworn I mentioned this before. I think there's a risk where we'll process a thing more than once because the Paginator will do multiple passes over issues if they are modified while the indexer is running. This is fine for actions because we filter them out, but it's not clear it works well for issues.

Reply:
I think you're missing some complication here. I tried to explain it in a comment at the top of the Paginator, but let me expand on it here.

The indexer works by finding all the issues that could changed within the time frame. It orders by them by updated date (descending), so new issues will pop to the beginning of the list. You query Jira's API by use of `start` and `limit` params: order all the issues in the query by updated date, skip the first M items and return the next N. The Jira API isn't stateful though; each time you query it, it generates that list anew.

There are two complications that can arise. The first is that a new issue can enter the list at the beginning. Imagine you're querying two at a time. You have issues A, B, C, D, E in order, and on the first pass you get A and B. On the second pass we'd want C and D. However someone has created a new issue F; the order is now F A B C D E; skipping two issues and returning the next two produces B C (because the first two are F A). No problem here.

The second complication is when someone modifies an existing issue from the end of the list and puts it at the beginning. We have the same A B C D E, and we start with A B. Then someone modifies E, so our list is E A B C D and we process B (skipping it because it hasn't changed) C. Then we process D. But now we need to start over to make sure we capture all the changes that have happened since we started working. So we process E A B C D. But while we're working, someone modifies E again. (Or really, any of the issues). Now we'll re-process it and re-write its actions to the TSV file.

See the comments on `getActionsFilterByLastSeen()` for more details.

Reply:
It's possible this won't matter because of the date filtering; I haven't fully processed out how all that works with jiraissues.

Reply:
If an issue was updated, that new action for the issue would fall out of the timerange unless we were querying the current date to tomorrow (is that what jiraactions does?) The actions that ARE in the time range would then be filtered out because we have already seen them, leaving filteredActions empty and jiraissues won't write it.